File: find_time.m
function [ timeInfo ] = find_time( data )
%find_time  It finds the time
%   Detailed explanation goes here

%Initialize timeInfo
timeInfo = zeros(1,4);

%Define hundreds, tens, and ones matrices
hundreds = [200 100];
tens = [80 40 20 10];
ones = [8 4 2 1];

%Determine year using the dot product
yearTens = dot(tens, data(46:49));
yearOnes = dot(ones, data(51,54));
year = yearTens + yearOnes;

%Determine day
dayHundreds = dot(hundreds, data(23,34));
dayTens = dot(tens, data(26:29));
dayOnes = dot(ones, data(31,34));
day = dayHundreds + dayTens + dayOnes;

%Determine hour
hourTens = dot(tens, data(11:14));
hourOnes = dot(ones, data(16,19));
hour = hourTens + hourOnes;

%Determine minute
minuteTens = [40 20 10] * data(2,4)';
minuteOnes = dot(ones, data(6,9));
minute = minuteTens + minuteOnes;

%Fill time info with values
timeInfo(1:4) = [year day hour minute];

end

File: find_time.c
void find_time(int *data, int *timeinfo) {
	//Given the received data, this finds the year, day, hour and minute
	
	//Determine the year
	int yearTens = 80*data[45] + 40*data[46] + 20*data[47] + 10*data[48];
	int yearOnes = 8*data[50] + 4*data[51] + 2*data[52] + data[53];
	int year = yearTens + yearOnes;
	
	//Determine the day
	int dayHundreds = 200*data[22] + 100*data[23];
	int dayTens = 80*data[25] + 40*data[26] + 20*data[27] + 10*data[28];
	int dayOnes = 8*data[30] + 4*data[31] + 2*data[32] + data[33];
	int day = dayHundreds + dayTens + dayOnes;
	
	//Determine the hour
	int hourTens = 80*data[10] + 40*data[11] + 20*data[12] + 10*data[13];
	int hourOnes = 8*data[15] + 4*data[16] + 2*data[17] + data[18];
	int hour = hourTens + hourOnes;
	
	//Determine the minute
	int minuteTens = 40*data[1] + 20*data[2] + 10*data[3];
	int minuteOnes = 8*data[5] + 4*data[6] + 2*data[7] + data[8];
	int minute = minuteTens + minuteOnes;
	
	timeinfo[0] = year;
	timeinfo[1] = day;
	timeinfo[2] = hour;
	timeinfo[3] = minute;
}	
